includes/classes/RegisteredDataHandler.php

```diff
@@ -0,0 +1,637 @@
+			'modified' => $modified,
+		);
+	}
+
+	/**
+	 * This function processes the registered data for the post content and post meta.
+	 * It calls the callback function provided in the registered data and updates the post data.
+	 *
+	 * @since x.x.x
+	 *
+	 * @param array $post_data The post data.
+	 * @param bool  $is_rest   Whether the post data is from the REST API.
+	 * @return array $post_data The processed post data.
+	 */
+	public function process_registered_data( $post_data, $is_rest = false ) {
+		// Filter is documented in includes/classes/DistributorPost.php
+		if ( ! apply_filters( 'dt_process_extra_data', true ) ) {
+			return $post_data;
+		}
+
+		// Get the registered data.
+		$registered_data = distributor_get_registered_data();
+
+		// Skip if no registered data is found.
+		if ( empty( $registered_data ) ) {
+			return $post_data;
+		}
+
+		foreach ( $registered_data as $data_key => $data ) {
+			$location    = $data['location'];
+			$attributes  = $data['attributes'];
+			$callback_fn = $data['post_distribute_cb'] ?? null;
+
+			// Skip if the callback function is not provided or not callable.
+			if ( empty( $callback_fn ) || ! is_callable( $callback_fn ) ) {
+				continue;
+			}
+
+			$extra_data = $post_data['distributor_extra_data'][ $data_key ] ?? array();
+
+			if ( 'post_meta' === $location ) {
+				$metadata_key = 'distributor_meta';
+				if ( isset( $post_data['meta'] ) && ! empty( $post_data['meta'] ) ) {
+					$metadata_key = 'meta';
+				}
+
+				$post_meta = $post_data[ $metadata_key ] ?? array();
+
+				if ( empty( $post_meta ) ) {
+					continue;
+				}
+
+				$post_data[ $metadata_key ] = $this->process_registered_post_meta_data( $post_meta, $data, $extra_data, $post_data );
+
+			} elseif ( 'post_content' === $location ) {
+				$content_key = 'post_content';
+				if ( $is_rest ) {
+					$content_key = 'content';
+					if ( isset( $post_data['distributor_raw_content'] ) ) {
+						$content_key = 'distributor_raw_content';
+					}
+				}
+
+				$post_content = $post_data[ $content_key ] ?? '';
+				$block_name   = $attributes['block_name'] ?? '';
+				$shortcode    = $attributes['shortcode'] ?? '';
+
+				if ( ( empty( $block_name ) && empty( $shortcode ) ) || empty( $post_content ) ) {
+					continue;
+				}
+
+				if ( ! empty( $block_name ) && has_blocks( $post_content ) ) {
+					$post_data[ $content_key ] = $this->process_registered_block_data( $post_content, $data, $extra_data, $post_data );
+					$post_content              = $post_data[ $content_key ];
+				}
+
+				// Process the shortcode if shortcode is provided.
+				if ( ! empty( $shortcode ) ) {
+					$post_data[ $content_key ] = $this->process_registered_shortcode_data( $post_content, $data, $extra_data, $post_data );
+				}
+			}
+		}
+
+		/**
+		 * Filter the post data after processing the registered data.
+		 *
+		 * @since x.x.x
+		 * @hook dt_after_registered_data_processed
+		 *
+		 * @param {array} $post_data       The post data.
```

Comment:
```suggestion
		 * @param {array} $post_data       The post data after processing.
```

It might be worth including the unmodified data as a parameter here (very, very optional as we can always add it later if requested).

:1234: Applies to other locations the filter is called.

Reply:
Handled in https://github.com/10up/distributor/pull/1309/commits/11d6ab5d8801a0d3c604be32b30c91c7a3f6b54c